barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_format.cpp

```diff
@@ -114,151 +79,136 @@ template <typename Builder> struct HonkRecursionConstraintsOutput {
 };
 
 template <typename Builder>
-void build_constraints(Builder& builder, AcirProgram& program, const ProgramMetadata& metadata)
+void build_constraints(Builder& builder, AcirFormat& constraints, const ProgramMetadata& metadata)
```

Comment:
unrelated to this PR specifically but its always annoyed me that we cant pass constraints as a const reference. Off the top of my head I think the impediments are (1) mutations to quad constraints and (2) the gate count tracking. (2) would be easy to solve but (1) might take more care. This isn't critical but to me the fact that we can't do it suggests a faulty design. Maybe somehting to just keep in mind as this work progresses